src/application.ts

```diff
@@ -107,7 +111,8 @@ export default class Application implements Owner {
     if (!mainTemplate) { throw new Error("Could not find main template."); }
 
     let mainLayout = templateFactory(mainTemplate).create(this.env);
-    let templateIterator = mainLayout.render(null, this.rootElement, new DynamicScope());
+    let rootRef = new UpdatableReference({ roots: this.roots });
+    let templateIterator = mainLayout.render(rootRef, this.rootElement, new DynamicScope());
```

Comment:
With that change, we can probably just render into `document.body` directly (since the template doesn't actually render anything, it doesn't really matter where it goes)